instagram/src/Authentication.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import LogIn from './LogIn';
+
+
+
+const Authentication = App => 
+class extends React.Component {
+    constructor (props) {
+        super(props);
+        this.state = {
+            LoggedIn: false,
+        }
+    }
+
+    componentDidMount(){
+       if (!localStorage.getItem('user')) {
+           this.setState({
+               LoggedIn:false,
+           })
+       }
+
+       else {
+        this.setState({
+            LoggedIn:true,
+        })
+       }
+    }
+
+    render() {
+        if (this.state.LoggedIn) return <App />;
```

Comment:
You can use a ternary statement here instead of a if statement. `this.state.LoggedIn ? <App /> : <Login />`